### DISCUSSION TITLE
Omitting scripting backend from editor window drawing and build folders

### DISCUSSION THREAD

**Discussion by nickfourtimes** [superunitybuild/buildtool#148] [Q&A] [answered]
Hey, (and apologies if there's a better place to ask these sorts of questions)

I'm adding some Build Platforms for a current project; in particular, they're console platforms, and so their scripting backend can _only_ be set to IL2CPP. Since this is a constant, I'd like to avoid seeing "Mono/IL2CPP" specified in the SUB editor window; likewise, I'd like to avoid that being a component of the build output folder hierarchy – e.g., where Windows might have `Debug\PC\Windows x64\IL2CPP\{game stuff}`, I'd like to simply have `Debug\{Platform}\{Architecture}\{game stuff}`.

However, if I omit setting up `scriptingBackends` in the build platform's `Init()` method, it looks like the build platform gets set to Mono by default, and still shows up in the above places (even though I think the builds are still being correctly forced to IL2CPP anyway).

Is there a way to get SUB to completely omit the scripting backend as a component of both the configuration names in the editor window, and in the build output folder hierarchy?

**Comment by robinnorth** [comment, marked as answer]
Hey @nickfourtimes, you've just prompted me to enable Discussions for the project so that there is a better place to ask these sorts of questions 🎉 

There currently isn't a way to omit the scripting backend from the build configuration key names, and although you can omit them from the build output path by removing the `$SCRIPTING_BACKEND` from the build path template, this obviously will cause a output path collisions if you enable Mono and IL2CPP for platforms that support both.

Are you looking to do this for aesthetic purposes, or for another reason?

**Reply by nickfourtimes** [reply]
Oh, lol, when I saw "Issue was converted into a Discussion," I thought I'd totally missed the Discussions tab. Glad to hear you'd just enabled it then!

Honestly this was purely aesthetic for the moment, just one extra folder I have to click into to find the build artifact. Not really a problem! Just wanted to make sure I wasn't missing anything obvious.

Thanks for the answer! And hoping to contribute more to SUB as we keep working with it.

**Reply by robinnorth** [reply]
You're most welcome! I feel you with the annoyance of navigating through build folders, but at the moment I'm prioritising getting the v8.0.0 release wrapped up, so I am unlikely to make any changes around this any time soon. Any contributions are always welcome though 🙏
